File: pruebas/docker-node/tcp_client.js
const net = require('net');

if (process.argv.length !== 4) {
    console.log('Usage: node tcp_client.js <host> <port>');
    process.exit(1);
}

const HOST = process.argv[2];
const PORT = parseInt(process.argv[3]);

console.log('Connecting to ' + HOST + ':' + PORT);

const client = new net.Socket();

client.on('connect', () => {
    console.log('Connected to ' + HOST + ':' + PORT);
    const message = 'hello\r\n';
    client.write(message.toString('utf-8')); // No es necesario convertir a UTF-8
});

client.on('data', data => {
    console.log('Received data: ' + data.toString('utf-8'));
    client.end();
});

client.on('close', () => {
    console.log('Client connection closed');
});

client.on('error', err => {
    console.error('Error: ' + err);
});

client.on('end', () => {
    console.log('Finished');
    client.destroy();
});

client.connect(PORT, HOST);

File: pruebas/tcp_client.py
import socket
import sys
import threading

    
def main(argv: list):
    if len(argv) != 3:
        print(f"Usage: {argv[0]} <host> <port>")
        sys.exit(1)
        
    HOST = argv[1]
    PORT = int(argv[2])
    try:
        client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        client_socket.connect((HOST, PORT))
        print(f"Connected to server at {HOST}:{PORT}")

        message = "Hello from client!"
        client_socket.send(message.encode('utf-8'))
        print(f"Sent: {message}")

        response = client_socket.recv(1024)
        print(f"Received: {response.decode('utf-8')}")

        client_socket.close()
        print("Connection closed")

    except KeyboardInterrupt:
        print("Ctrl + C pressed. Stopping client...")
        client_socket.close()
        print("Client stopped.")

if __name__ == "__main__":
    main(sys.argv)
